### DISCUSSION TITLE
Integrating Undetected-Chromedriver with SeleniumBase to Load CRX Extensions

### DISCUSSION THREAD

**Discussion by benfrankie** [seleniumbase/SeleniumBase#2763] [Q&A] [answered]
Hello,

I am currently using SeleniumBase for my testing needs and have been leveraging the undetected-chromedriver (uc) for anti-detection capabilities. However, I am facing issues when trying to load a Chrome extension (CRX file) while using the undetected mode provided by SeleniumBase.

Here's a snippet of what I tried that does not seem to work:

from seleniumbase import Driver

extension_path = r'C:\Path\To\Extension\Urban-VPN-Proxy.crx'

driver = Driver(
    uc=True,
    extension_zip=extension_path,
    undetectable=True,
)

This method fails to load the extension, whereas using the standard Selenium approach with ChromeOptions to add the extension works fine:

from selenium import webdriver
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.chrome.service import Service

options = webdriver.ChromeOptions()
options.add_extension(extension_path)

service = Service(executable_path=ChromeDriverManager().install())
driver = webdriver.Chrome(service=service, options=options)

Could you provide guidance on how to properly integrate CRX extensions with undetected-chromedriver within the SeleniumBase framework? 

Thank you for your help!

**Comment by mdmintz** [comment, marked as answer]
That's been a limitation since the original undetected-chromedriver, and even the maintainer of that couldn't figure it out:
https://github.com/ultrafunkamsterdam/undetected-chromedriver/issues?q=is%3Aissue+is%3Aopen+CRX

If you can unzip the extension to a dir, you can pass it in as `extension_dir`. That works.

**Reply by benfrankie** [reply]
The extension encounters an error when installing from a zip file or as unpacked. It seems that using this extension is only possible by installing it from the webstore or with a .crx file. As a workaround, I am attempting to automate the process of installing a Chrome extension by clicking the 'Add to Chrome' button and interacting with the confirmation popup. Could you provide guidance or suggest a method within SeleniumBase to click the 'Add extension' button on the popup?

I tried the code below, but it doesn't focus on the popup:
action = ActionChains(driver)
action.send_keys(Keys.TAB)
action.send_keys(Keys.ENTER)
action.perform()


Thank you!
